File: 389. 找不同/389. 找不同.java




class Solution {
    public char findTheDifference(String s, String t) {
        int[] l = new int[26];
        for (int i = 0; i < s.length(); i++) {
            l[(byte)s.charAt(i)-(byte)'a'] ++;
        }
        for (int i = 0; i < t.length(); i++) {
            l[(byte)t.charAt(i)-(byte)'a'] --;
            if(l[(byte)t.charAt(i)-(byte)'a']==-1){
                return t.charAt(i);
            }

        }
        return 'a';
    }
}

File: 389. 找不同/389. 找不同.py


class Solution:
    def findTheDifference(self, s, t):
        l = [0] * 26
        for c in s:
            l[ord(c)-ord('a')] += 1
        for c in t:
            l[ord(c)-ord('a')] -= 1
            if(l[ord(c)-ord('a')]) == -1:
                return c
            
            
            

s = Solution()
print(s.findTheDifference('s','sr'))

